go-controller/vendor/github.com/coreos/go-iptables/iptables/iptables.go

```diff
@@ -45,14 +45,21 @@ func (e *Error) Error() string {
 	return fmt.Sprintf("running %v: exit status %v: %v", e.cmd.Args, e.ExitStatus(), e.msg)
 }
 
+var isNotExistPatterns = []string{
+	"Bad rule (does a matching rule exist in that chain?).\n",
+	"No chain/target/match by that name.\n",
+	"No such file or directory",
+	"does not exist",
+}
+
 // IsNotExist returns true if the error is due to the chain or rule not existing
 func (e *Error) IsNotExist() bool {
```

Comment:
ah we have a function here but we don't use it in our code to see if the chain exists or not?

Reply:
I guess so, maybe thats because it would be another iptables bash call and probalby take even longer :)